### ISSUE TITLE
Check TRANSFEREXPORT_RETRIEVE for plain clones

### ISSUE THREAD

**Issue by mih** [datalad/datalad-dataverse#239]
With #237 it refuses any download without a fileid on record.

```py
    def transferexport_retrieve(self, key, local_file, remote_file):
        stored_ids = self._get_annex_fileid_record(key)
        if not stored_ids:
            raise RemoteError(f"Key {key} unavailable")
```

This is inconsistent with `CHECKPRESENTEXPORT`. In #237 I added a comment on that. Needs an investigation and uniform handling.

**Comment by bpoldrack** [comment]
To have it on record here:

The inconsistency is currently about falling back to plain matching when there is no ID on record in order to be able to retrieve the datalad-annex keys when cloning from an export remote. During that cloning there's no repository yet, that can have the ID for those keys on record (we are only about to receive the respective git-annex branch). At least that is the rationale for checkpresentexport. Originally, that was therefore special casing those keys. 

Need to double-check whether that reasoning (still) applies.

**Comment by mih** [comment]
I fixed this in ac59f8e02ee343eadb9b1878a831a141b07cfdfc and posted https://github.com/datalad/datalad-dataverse/issues/246 as a TODO
